packages/core/lib/middleware/mw_utils.js

```diff
@@ -134,6 +138,56 @@ var utils = {
       throw new Error('Please specify a path to the local sampling rules file, or supply an object containing the rules.');
 
     this.sampler.setLocalRules(source);
+  },
+
+  /**
+   * Traces the request/response cycle of an http.IncomingMessage / http.ServerResponse pair.
+   * Resolves sampling rules, creates a segment, adds the core request / response data adding
+   * throttling / error / fault flags based on the response status code.
+   * @param {http.IncomingMessage} req - The incoming request.
+   * @param {http.ServerResponse} res - The server response.
+   * @returns {Segment}
+   * @memberof AWSXRay
+   */
+  traceRequestResponseCycle: function traceRequestResponseCycle(req, res) {
+    var amznTraceHeader = this.processHeaders(req);
+    var name = this.resolveName(req.headers.host);
+    var segment = new Segment(name, amznTraceHeader.root, amznTraceHeader.parent);
+
+    var responseWithEmbeddedRequest = Object.assign({}, res, { req: req });
```

Comment:
The `resolveSampling()` function relies on `res` having an embedded `req` object. This is not on a vanilla NodeJS `http.ServerResponse` so we have to add it here (https://nodejs.org/api/http.html#http_class_http_serverresponse)